app.py

```diff
@@ -30,15 +30,29 @@ def location_page(location_id, rack_id):
     data = sql_queries.get_table_size(location_id)
 
     with open('cellsPerBay.json') as json_file:
-        cells_per_bay = json.load(json_file)
+        data2 = json.load(json_file)
 
+    # Check to see if location is in file
+    if(location_id in data2):
+        cells_per_bay = data2[location_id]
+    else:
+        return render_template('errors/404.html'), 404
+
+    # Check if the url is valid
     tableType = check_table_type(data["cell"])
+    if(tableType == "3d"):
+        if(int(rack_id) > data["tableSize"][0]):
+            return render_template('errors/404.html'), 404
+    else:
+        if(rack_id is not None):
+            return render_template('errors/404.html'), 404
```

Comment:
Try using the _flash_ message feature of flask to inform the user that the location id/rack number/wrong table type rather than redirecting them to a 404 page. Also you don' need brackets around your conditional statements.